presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -322,11 +324,41 @@ private static void process(QueryRunner queryRunner, String sql, OutputFormat ou
             queryRunner.setSession(session);
         }
         catch (RuntimeException e) {
-            System.err.println("Error running command: " + e.getMessage());
-            if (queryRunner.getSession().isDebug()) {
-                e.printStackTrace();
+            System.err.println(getErrorMessage(e, queryRunner.getSession()));
+        }
+    }
+
+    private static String getErrorMessage(RuntimeException e, ClientSession session)
+    {
+        StringWriter sw = new StringWriter();
+        e.printStackTrace(new PrintWriter(sw));
+        String stackTrace = sw.toString();
+
+        StringBuilder builder = new StringBuilder();
+        if (e.getCause() instanceof java.io.EOFException) {
+            builder.append("There was a problem with a response from Presto Coordinator." + "\n");
+            builder.append("To solve this problem you may try to:\n");
+            builder.append(" * Verify that Presto is running on " + session.getServer() + "\n");
+            builder.append(" * Use '--server' argument when starting Presto CLI to define server host and port.\n");
+            builder.append(" * Check the network conditions between client and server.\n");
+            if (!session.isDebug()) {
+                builder.append(" * Use '--debug' argument to get more technical details.\n");
             }
         }
+        else {
+            builder.append("Error running command: " + e.getMessage() + "\n");
+        }
+
+        if (session.isDebug()) {
+            builder.append("\n=========   TECHNICAL DETAILS   =========\n");
```

Comment:
I'd rather not add this since this is debug mode, where you expect "technical details" to be printed (otherwise you wouldn't have specified `--debug`).